### PR TITLE
Revising memory management landing page

### PR DESCRIPTION
**Pull request by kaitlynmichael**
DOC-1064
staging: https://docs.redis.com/staging/jira-doc-1064/rs/concepts/memory-performance/

### CODE REVIEW

**Review by lanceleonard — APPROVED** [review]
LGTM

File: content/rs/concepts/memory-performance/_index.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+## Database persistence
+
+Both RAM memory and flash memory are at risk of data loss if a server or process fails. Persisting your data to disk helps protect it against loss in those situations. You can configure persistence at the time of database creation, or by editing the database’s configuration.
+
+There are two main types of persistence strategies in Redis Enterprise Software: append-only files (AoF) and snapshots.
+
+Append-only files (AoF) keep a record of data changes and writes each change to the end of a file, allowing you to recover the dataset by replaying the writes in the append-only log.
+
+Snapshots capture all the data as it exists in one moment in time and writes it to disk, allowing you to recover the entire data set as it existed at that moment in time.
```
Comment (rrelledge) [inline content/rs/concepts/memory-performance/_index.md:52]:
```suggestion
Snapshots capture all the data as it exists in one moment in time and writes it to disk, allowing you to recover the entire dataset as it existed at that moment in time.
```

File: content/rs/concepts/memory-performance/_index.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+For more info on data persistence see [Database persistence with Redis Enterprise Software]({{<relref "/rs/concepts/memory-performance/persistence.md">}}) or [Durable Redis](https://redis.com/redis-enterprise/technology/durable-redis/).
+
+## Redis on Flash (RoF)
+
+By default, Redis Enterprise Software stores your data entirely in [RAM](https://en.wikipedia.org/wiki/Random-access_memory) for improved performance. The [Redis on Flash]({{<relref "/rs/concepts/memory-performance/redis-flash.md">}}) feature enables your data to span both RAM and [SSD](https://en.wikipedia.org/wiki/Solid-state_drive) storage ([flash memory](https://en.wikipedia.org/wiki/Flash_memory)). Keys are always stored in RAM, but Redis on Flash manages the location of their values. Frequently used (hot) values are stored on RAM, but infrequently used (warm) values are moved to flash memory. This saves on expensive RAM space, which give you comparable performance at a lower cost for large data sets.
```
Comment (rrelledge) [inline content/rs/concepts/memory-performance/_index.md:58]:
```suggestion
By default, Redis Enterprise Software stores your data entirely in [RAM](https://en.wikipedia.org/wiki/Random-access_memory) for improved performance. The [Redis on Flash]({{<relref "/rs/concepts/memory-performance/redis-flash.md">}}) feature enables your data to span both RAM and [SSD](https://en.wikipedia.org/wiki/Solid-state_drive) storage ([flash memory](https://en.wikipedia.org/wiki/Flash_memory)). Keys are always stored in RAM, but Redis on Flash manages the location of their values. Frequently used (hot) values are stored on RAM, but infrequently used (warm) values are moved to flash memory. This saves on expensive RAM space, which give you comparable performance at a lower cost for large datasets.
```

File: content/rs/concepts/memory-performance/_index.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+For more info, see [Redis on Flash]({{<relref "/rs/concepts/memory-performance/redis-flash.md">}}).
+
+## Shard placement
+
+The location of the primary and replica shards on the cluster nodes can impact your database performance.
+primary shards and their corresponding replica shards are always placed on separate nodes for data resiliency.
```
Comment (rrelledge) [inline content/rs/concepts/memory-performance/_index.md:65]:
```suggestion
Primary shards and their corresponding replica shards are always placed on separate nodes for data resiliency.
```

File: content/rs/concepts/memory-performance/_index.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+Redis Enterprise Software has two shard placement policies available:
+
+- **dense**: puts as many shards as possible on the smallest number of nodes
+- **sparse**: spread the shards across as many nodes as possible
+
+For more info about the shard placement policy, see [Shard placement policy]({{<relref "/rs/concepts/memory-performance/shard-placement-policy.md">}})
+
+## Metrics
+
+From the Redis Enterprise Software admin console, you can monitor the performance of you clusters, nodes, databases, and shards with real-time metrics. You can also enable alerts for node, cluster, or database events such as high memory usage or throughput.
```
Comment (rrelledge) [inline content/rs/concepts/memory-performance/_index.md:77]:
```suggestion
From the Redis Enterprise Software admin console, you can monitor the performance of your clusters, nodes, databases, and shards with real-time metrics. You can also enable alerts for node, cluster, or database events such as high memory usage or throughput.
```

File: content/rs/release-notes/rs-6-0-8-september-2020.md [hunk 5]
```diff
@@ -71,7 +71,7 @@ With build 6.0.8-32:
 
 ### Active-Active databases
 - RS44656 - A bug causing TLS mode for clients connections to toggle between ‘all communication’ to ‘for crdb communication only’ when performing a global configuration change. ***TBD***
-- RS51359 - Active-Active databases, using replication and Append Only File (AOF) for [Database Persistence]({{< relref "rs/concepts/data-access/persistence.md" >}}) are suffering from memory leak on slave the shards causing them to grow bigger than the master shards. Customers are advised to upgrade to RS 6.0.12 ***TBD***. Meanwhile you can use snapshot for Database Persistence or restart the slave shards ***TBD***.
+- RS51359 - Active-Active databases, using replication and Append Only File (AOF) for [Database persistence]({{< relref "/rs/concepts/memory-performance/persistence.md" >}}) are suffering from memory leak on slave the shards causing them to grow bigger than the master shards. Customers are advised to upgrade to RS 6.0.12 ***TBD***. Meanwhile you can use snapshot for Database Persistence or restart the slave shards ***TBD***.
```
Comment (rrelledge) [inline content/rs/release-notes/rs-6-0-8-september-2020.md:74]:
```suggestion
- RS51359 - Active-Active databases, using replication and Append Only File (AOF) for [Database persistence]({{< relref "/rs/concepts/memory-performance/persistence.md" >}}), are suffering from memory leaks on replica shards, causing them to grow bigger than the master shards. Customers are advised to upgrade to RS 6.0.12 ***TBD***. Meanwhile you can use snapshots for Database Persistence or restart the replica shards ***TBD***.
```

File: content/rs/release-notes/rs-6-0-may-2020.md [hunk 6]
```diff
@@ -107,7 +107,7 @@ To use the updated modules with a database, you must [upgrade the module on the
 ## Known limitations
 
 ### Active-Active databases
-- RS51359 - Active-Active databases, using replication and Append Only File (AOF) for [Database Persistence]({{< relref "rs/concepts/data-access/persistence.md" >}}) are suffering from memory leak on slave the shards causing them to grow bigger than the master shards. Customers are advised to upgrade to RS 6.0.12 ***TBD***. Meanwhile you can use snapshot for Database Persistence or restart the slave shards ***TBD***.
+- RS51359 - Active-Active databases, using replication and Append Only File (AOF) for [database persistence]({{< relref "rs/concepts/memory-performance/persistence.md" >}}) are suffering from memory leak on slave the shards causing them to grow bigger than the master shards. Customers are advised to upgrade to RS 6.0.12 ***TBD***. Meanwhile you can use snapshot for database persistence or restart the slave shards ***TBD***.
```
Comment (rrelledge) [inline content/rs/release-notes/rs-6-0-may-2020.md:110]:
```suggestion
- RS51359 - Active-Active databases, using replication and Append Only File (AOF) for [database persistence]({{< relref "rs/concepts/memory-performance/persistence.md" >}}), are suffering from memory leaks on replica shards, causing them to grow bigger than the master shards. Customers are advised to upgrade to RS 6.0.12 ***TBD***. Meanwhile you can use snapshots for database persistence or restart the replica shards ***TBD***.
```

File: content/rs/administering/designing-production/performance/optimization.md [hunk 7]
```diff
@@ -1,40 +1,45 @@
...
+performance. As part of this process, Redis Enterprise Software examines usage characteristics
+and load to adjust its run-time configuration accordingly. Depending
+on your specific usage characteristics and load, it might take Redis Enterprise Software some
 time to adjust itself to optimal performance.
 
 To ensure optimal performance, you must run your workload several times
 and for a long duration until performance stabilizes.
 
-In addition, RS can be optimized for two different environments:
+In addition, you can be optimize your cluster for two different environments:
```
Comment (rrelledge) [inline content/rs/administering/designing-production/performance/optimization.md:22]:
```suggestion
In addition, you can optimize your cluster for two different environments:
```

File: content/rs/administering/database-operations/updating-configurations.md [hunk 8]
```diff
@@ -1,5 +1,5 @@
 ---
-Title: Updating the Database Configuration
+Title: Updating the database configuration
```
Comment (rrelledge) [inline content/rs/administering/database-operations/updating-configurations.md:2]:
```suggestion
Title: Update the database configuration
```

File: content/rs/concepts/memory-performance/_index.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+There are a number of factors to consider when sizing your database:
+
+- **dataset size**: you want your limit to be above your dataset size to leave room for overhead.
+- **database throughput**: high throughput needs a higher memory limit.
+- [**modules**]({{<relref "/modules/_index.md">}}): using modules with your database consumes more memory.
+- [**database clustering**]({{<relref "/rs/concepts/high-availability/clustering.md">}}): spreading your data into shards across multiple nodes means you cannot disable clustering or reduce the number of shards later.
+- [**database replication**]({{<relref "/rs/concepts/high-availability/replication.md">}}): enabling replication doubles memory consumption
+- [**Active-Active replication**]({{<relref "/rs/administering/designing-production/active-active.md">}}): enabling Active-Active replication requires double the memory of regular replication, which can be up to four times(4x) the original data size.
```
Comment (rrelledge) [inline content/rs/concepts/memory-performance/_index.md:32]:
```suggestion
- [**Active-Active replication**]({{<relref "/rs/administering/designing-production/active-active.md">}}): enabling Active-Active replication requires double the memory of regular replication, which can be up to four times (4x) the original data size.
```

**Review by rrelledge — APPROVED** [review]
Looks good to me. I just left a couple of suggestions.

File: content/rs/concepts/memory-performance/_index.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    /rs/concepts/memory-performance/_index/,
+
+]
+---
+Redis Enterprise Software has multiple mechanisms in its
+architecture to help optimize storage and performance.
+
+## Memory limits
+
+Database memory limits define the maximum size your database can reach across all database replicas and [shards]({{<relref "rs/concepts/terminology.md#redis-instance-shard">}}) on the cluster. This limit includes data values, keys, module data, and overhead for other features. Your memory limit will also determine the number of shards you'll need.
```
Comment (AlonMagrafta) [inline content/rs/concepts/memory-performance/_index.md:23]: I think we need to be clear here about Database memory limits include:
- Replication (if enabled)
- A/A overhead (if enabled)

File: content/rs/concepts/memory-performance/_index.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+There are a number of factors to consider when sizing your database:
+
+- **dataset size**: you want your limit to be above your dataset size to leave room for overhead.
+- **database throughput**: high throughput needs a higher memory limit.
```
Comment (AlonMagrafta) [inline content/rs/concepts/memory-performance/_index.md:28]: That's not that accurate.
The correct logic is higher throughput requires higher number of shards > that provides higher memory.

File: content/rs/concepts/memory-performance/_index.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+- [**modules**]({{<relref "/modules/_index.md">}}): using modules with your database consumes more memory.
```
Comment (AlonMagrafta) [inline content/rs/concepts/memory-performance/_index.md:29]: I think we should have @K-Jo 's comment on that one

File: content/rs/concepts/memory-performance/_index.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+- [**database clustering**]({{<relref "/rs/concepts/high-availability/clustering.md">}}): spreading your data into shards across multiple nodes means you cannot disable clustering or reduce the number of shards later.
```
Comment (AlonMagrafta) [inline content/rs/concepts/memory-performance/_index.md:30]: Terminology wise - we are saying 'scaling out' is possible but 'scaling in' is not (aka horizontal scaling).
.
For general knowledge, the other method is scaling up and down (aka Vertical scaling)
.
The main difference between scaling up and scaling out is that horizontal scaling simply adds more machine resources to your existing machine infrastructure. Vertical scaling adds power to your existing machine infrastructure by increasing power from CPU or RAM to existing machines

File: content/rs/concepts/memory-performance/_index.md [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+- [**database replication**]({{<relref "/rs/concepts/high-availability/replication.md">}}): enabling replication doubles memory consumption
+- [**Active-Active replication**]({{<relref "/rs/administering/designing-production/active-active.md">}}): enabling Active-Active replication requires double the memory of regular replication, which can be up to four times (4x) the original data size.
+
+For more information on memory limits, see [Memory management with Redis Enterprise Software]({{<relref "/rs/concepts/memory-performance/memory-management.md">}}) or [Database memory limits]({{<relref "/rs/concepts/memory-performance/memory-limit.md">}}).
+
+## Eviction policies
+
+When a database exceeds its memory limit, eviction policies determine which data is removed. The eviction policy removes keys based on frequency of use, how recently used, randomly, expiration date, or a combination of these factors. The policy can also be set to `noeviction` to return a memory limit error when trying to insert more data.
+
+The default eviction policy for databases is `volatile-lru` which evicts the least recently used keys out of all keys with the ‘expire’ field set.
```
Comment (AlonMagrafta) [inline content/rs/concepts/memory-performance/_index.md:40]: we should add the default for AA DBs is 'noeviction'

File: content/rs/concepts/memory-performance/_index.md [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+## Redis on Flash (RoF)
+
+By default, Redis Enterprise Software stores your data entirely in [RAM](https://en.wikipedia.org/wiki/Random-access_memory) for improved performance. The [Redis on Flash]({{<relref "/rs/concepts/memory-performance/redis-flash.md">}}) feature enables your data to span both RAM and [SSD](https://en.wikipedia.org/wiki/Solid-state_drive) storage ([flash memory](https://en.wikipedia.org/wiki/Flash_memory)). Keys are always stored in RAM, but Redis on Flash manages the location of their values. Frequently used (hot) values are stored on RAM, but infrequently used (warm) values are moved to flash memory. This saves on expensive RAM space, which give you comparable performance at a lower cost for large datasets.
+
+For more info, see [Redis on Flash]({{<relref "/rs/concepts/memory-performance/redis-flash.md">}}).
+
+## Shard placement
+
+The location of the primary and replica shards on the cluster nodes can impact your database performance.
+Primary shards and their corresponding replica shards are always placed on separate nodes for data resiliency.
```
Comment (AlonMagrafta) [inline content/rs/concepts/memory-performance/_index.md:65]:
```suggestion
Primary shards and their corresponding replica shards are always placed on separate nodes for data resiliency (High Availability).
```

**Pull request merged**